src/controllers/applicant.ts

```diff
@@ -22,31 +24,46 @@ export const createApplicant = async (req: Request, res: Response): Promise<void
     }
 };
 
-export const updateApplicantStatus = async (req: Request, res: Response): Promise<void> => {
-    try {
-        const valid_status = ['Pending', 'Accepted', 'Rejected', 'Scheduled'];
-        const my_applicant = await Applicant.findOne({ _id: req.body.id });
+export const updateApplicantFields = async (req: Request, res: Response): Promise<void> => {
+    const valid_status = [
+        'Pending',
+        'Screened: Accepted',
+        'Screened: Rejected',
+        'Scheduled for Interview',
+        'Final Decision: Accepted',
+        'Final Decision: Rejected',
+        'Archived: Rejected',
+    ];
+    const valid_level = ['Beginner', 'Intermediate', 'Advanced'];
+    const allowed_updates = ['id', 'status', 'level']; // id is passed in the body but is not an update
```

Comment:
We should change the API route to PATCH applicant/:applicantId - I don't think we should open the opportunity for somebody to change an applicant ID that would be pretty chaotic.

Reply:
I've applied the requested changes